### ISSUE TITLE
Poetry fails to install torch when running under python 3.11.0+

### ISSUE THREAD

**Issue by vladyslav-burylov** [python-poetry/poetry#7237]
- **Poetry version**: 1.3.1 (1.2.0 fails as well, I haven't tested other versions)
- **Python version**: 3.11.0 (and onwards)
- **OS version and name**: macOS 12.6 (both Intel and Apple M1 versions affected); fails inside Docker as well
- **pyproject.toml**: see attached output which cats config files

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

Hi team, `poetry 1.3.1` fails to install `torch 1.2.1 (or newer)` when running under `python 3.11.0 (or newer)`

It raises following error: `Unable to find installation candidates for torch (1.12.1)`

Any ideas how to fix that would be appreciated,
Thanks in advance!

## Details
- Tested on `macOs 12.6 inside virtual envs - both Intel and Apple M1 affected`
- Tested inside `Docker Desktop running on macOs - both Intel and Apple M1 affected`
- Everything seems to be `working well when running under python 3.10.9 or older`

## Steps to reproduce
- Make sure you have docker installed
- Try to build image from the following Dockerfile:
```Dockerfile
FROM python:3.11.0

# Install poetry
RUN pip install pip==22.3.1 setuptools==65.5.0 wheel==0.38.4 && \
    pip install poetry==1.3.1 && \
    pip cache purge

WORKDIR /opt/app

# Create config files
RUN echo '[virtualenvs]' > poetry.toml && \
    echo 'in-project = true' >> poetry.toml && \
    echo '' >> poetry.toml && \
    \
    echo '[build-system]' > pyproject.toml && \
    echo 'requires = ["poetry-core>=1.3.1"]' >> pyproject.toml && \
    echo 'build-backend = "poetry.core.masonry.api"' >> pyproject.toml && \
    echo '' >> pyproject.toml && \
    echo '[tool.poetry]' >> pyproject.toml && \
    echo 'name = "test"' >> pyproject.toml && \
    echo 'description = "test"' >> pyproject.toml && \
    echo 'authors = ["test"]' >> pyproject.toml && \
    echo 'version = "0.0.1"' >> pyproject.toml && \
    echo '' >> pyproject.toml && \
    echo '[tool.poetry.dependencies]' >> pyproject.toml && \
    echo 'python = ">=3.10.9"' >> pyproject.toml && \
    echo 'torch = "=1.12.1"' >> pyproject.toml && \
    echo '' >> pyproject.toml

# Echo poetry debug info
RUN poetry debug info

# Echo config
RUN echo 'poetry.toml' && \
    echo '-----------' && \
    cat poetry.toml && \
    echo '' && \
    echo 'pyproject.toml' && \
    echo '--------------' && \
    cat pyproject.toml

# Install dependencies
RUN poetry install -vvv
```
### Expected Behavior
- poetry install succeeds

### Actual Behavior
- poetry install fails with `Unable to find installation candidates for torch (1.12.1)` error
- full dump of the output can be found below (it dumps pyproject.toml as well)
```shell
> docker build . --progress=plain
#1 [internal] load build definition from Dockerfile
#1 sha256:e972e83811363877c6e3920e4929012c80dc1f083a8da12d286f6ee59b62aee6
#1 transferring dockerfile: 1.38kB done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:d5d5449213620bbb25d5c2e07dc6576b544d463eb93d258f8df4d5e9bea6a148
#2 transferring context: 2B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/python:3.11.0
#3 sha256:9316d185864b8b0db1808fb040ddf8351c8e66f751eb2b3b3fe5a985625a0098
#3 DONE 1.9s

#4 [1/7] FROM docker.io/library/python:3.11.0@sha256:55101c8373de5311b0ece8746251882ddb0bb394588c93d2477b88852ad00e72
#4 sha256:1b328d15e97dd11ac3bef2d27691773d1f8b644ae713affbac0bb015ceba75ea
#4 DONE 0.0s

#5 [2/7] RUN pip install pip==22.3.1 setuptools==65.5.0 wheel==0.38.4 &&     pip install poetry==1.3.1 &&     pip cache purge
#5 sha256:5da28ee606cc7e60c399f9b041f6a4c5866df1b673b6be1f213d136e91056a94
#5 CACHED

#6 [3/7] WORKDIR /opt/app
#6 sha256:e8083d2108f3668fc8cbc273fb8eae7411c7a02862637c9680cf0de9a74385d0
#6 CACHED

#7 [4/7] RUN echo '[virtualenvs]' > poetry.toml &&     echo 'in-project = true' >> poetry.toml &&     echo '' >> poetry.toml &&         echo '[build-system]' > pyproject.toml &&     echo 'requires = ["poetry-core>=1.3.1"]' >> pyproject.toml &&     echo 'build-backend = "poetry.core.masonry.api"' >> pyproject.toml &&     echo '' >> pyproject.toml &&     echo '[tool.poetry]' >> pyproject.toml &&     echo 'name = "test"' >> pyproject.toml &&     echo 'description = "test"' >> pyproject.toml &&     echo 'authors = ["test"]' >> pyproject.toml &&     echo 'version = "0.0.1"' >> pyproject.toml &&     echo '' >> pyproject.toml &&     echo '[tool.poetry.dependencies]' >> pyproject.toml &&     echo 'python = ">=3.10.9"' >> pyproject.toml &&     echo 'torch = "=1.12.1"' >> pyproject.toml &&     echo '' >> pyproject.toml
#7 sha256:7e5090eaa0f229ee344a9b8b82b3aeef88a95e0e7f97339e17c7f60eab3bb5d0
#7 CACHED

#8 [5/7] RUN poetry debug info
#8 sha256:925fc9e554abcffd8ede12a4b7bd34bcc6d20ca71c5be1095d49cb3ffedfd766
#8 0.364
#8 0.364 Poetry
#8 0.401 Version: 1.3.1
#8 0.401 Python:  3.11.0
#8 0.402
#8 0.402 Virtualenv
#8 0.402 Python:         3.11.0
#8 0.402 Implementation: CPython
#8 0.402 Path:           NA
#8 0.402 Executable:     NA
#8 0.402
#8 0.429 System
#8 0.429 Platform:   linux
#8 0.429 OS:         posix
#8 0.429 Python:     3.11.0
#8 0.429 Path:       /usr/local
#8 0.429 Executable: /usr/local/bin/python3.11
#8 DONE 0.5s

#9 [6/7] RUN echo 'poetry.toml' &&     echo '-----------' &&     cat poetry.toml &&     echo '' &&     echo 'pyproject.toml' &&     echo '--------------' &&     cat pyproject.toml
#9 sha256:97ed3bd601f19b2c6cf9d41657e73618ac5bb71f36c6c1d5aeb86c3ebc9c21dd
#9 0.288 poetry.toml
#9 0.288 -----------
#9 0.289 [virtualenvs]
#9 0.289 in-project = true
#9 0.289
#9 0.289
#9 0.289 pyproject.toml
#9 0.289 --------------
#9 0.290 [build-system]
#9 0.290 requires = ["poetry-core>=1.3.1"]
#9 0.290 build-backend = "poetry.core.masonry.api"
#9 0.290
#9 0.290 [tool.poetry]
#9 0.290 name = "test"
#9 0.290 description = "test"
#9 0.290 authors = ["test"]
#9 0.290 version = "0.0.1"
#9 0.290
#9 0.290 [tool.poetry.dependencies]
#9 0.290 python = ">=3.10.9"
#9 0.290 torch = "=1.12.1"
#9 0.290
#9 DONE 0.3s

#10 [7/7] RUN poetry install -vvv
#10 sha256:e604398426d979314175f5b69da61a87a5368d0ebecf167170fddb028281c502
#10 0.461 Loading configuration file /opt/app/poetry.toml
#10 0.464 Creating virtualenv test in /opt/app/.venv
#10 0.696 Using virtualenv: /opt/app/.venv
#10 0.764 Updating dependencies
#10 0.764 Resolving dependencies...
#10 0.765    1: fact: test is 0.0.1
#10 0.765    1: derived: test
#10 0.766    1: fact: test depends on torch (=1.12.1)
#10 0.766    1: selecting test (0.0.1)
#10 0.766    1: derived: torch (==1.12.1)
#10 0.989    1: fact: torch (1.12.1) depends on typing-extensions (*)
#10 0.990    1: selecting torch (1.12.1)
#10 0.990    1: derived: typing-extensions
#10 1.184    1: selecting typing-extensions (4.4.0)
#10 1.184    1: Version solving took 0.419 seconds.
#10 1.184    1: Tried 1 solutions.
#10 1.196
#10 1.196 Writing lock file
#10 1.197
#10 1.197 Finding the necessary packages for the current system
#10 1.217
#10 1.217 Package operations: 2 installs, 0 updates, 0 removals
#10 1.217
#10 1.217   • Installing typing-extensions (4.4.0)
#10 1.934   • Installing torch (1.12.1)
#10 1.954
#10 1.954   Stack trace:
#10 1.954
#10 1.954   5  /usr/local/lib/python3.11/site-packages/poetry/installation/executor.py:263 in _execute_operation
#10 1.966       261│
#10 1.966       262│             try:
#10 1.966     → 263│                 result = self._do_execute_operation(operation)
#10 1.966       264│             except EnvCommandError as e:
#10 1.966       265│                 if e.e.returncode == -2:
#10 1.966
#10 1.966   4  /usr/local/lib/python3.11/site-packages/poetry/installation/executor.py:334 in _do_execute_operation
#10 1.976       332│             return 0
#10 1.976       333│
#10 1.976     → 334│         result: int = getattr(self, f"_execute_{method}")(operation)
#10 1.976       335│
#10 1.976       336│         if result != 0:
#10 1.976
#10 1.976   3  /usr/local/lib/python3.11/site-packages/poetry/installation/executor.py:454 in _execute_install
#10 1.986       452│
#10 1.986       453│     def _execute_install(self, operation: Install | Update) -> int:
#10 1.986     → 454│         status_code = self._install(operation)
#10 1.986       455│
#10 1.986       456│         self._save_url_reference(operation)
#10 1.986
#10 1.986   2  /usr/local/lib/python3.11/site-packages/poetry/installation/executor.py:488 in _install
#10 1.995       486│             archive = self._download_link(operation, Link(package.source_url))
#10 1.995       487│         else:
#10 1.995     → 488│             archive = self._download(operation)
#10 1.995       489│
#10 1.995       490│         operation_message = self.get_operation_message(operation)
#10 1.995
#10 1.995   1  /usr/local/lib/python3.11/site-packages/poetry/installation/executor.py:626 in _download
#10 2.005       624│
#10 2.005       625│     def _download(self, operation: Install | Update) -> Path:
#10 2.005     → 626│         link = self._chooser.choose_for(operation.package)
#10 2.005       627│
#10 2.005       628│         if link.yanked:
#10 2.005
#10 2.005   RuntimeError
#10 2.005
#10 2.005   Unable to find installation candidates for torch (1.12.1)
#10 2.005
#10 2.005   at /usr/local/lib/python3.11/site-packages/poetry/installation/chooser.py:105 in choose_for
#10 2.008       101│
#10 2.008       102│             links.append(link)
#10 2.008       103│
#10 2.008       104│         if not links:
#10 2.008     → 105│             raise RuntimeError(f"Unable to find installation candidates for {package}")
#10 2.008       106│
#10 2.008       107│         # Get the best link
#10 2.008       108│         chosen = max(links, key=lambda link: self._sort_key(package, link))
#10 2.008       109│
#10 2.008
#10 ERROR: executor failed running [/bin/sh -c poetry install -vvv]: exit code: 1
------
 > [7/7] RUN poetry install -vvv:
------
executor failed running [/bin/sh -c poetry install -vvv]: exit code: 1
```

**Comment by dimbleby** [comment]
because no distribution of torch 1.12.1 is available for python 3.11

https://pypi.org/project/torch/1.12.1/#files

not a bug

**Comment by vladyslav-burylov** [comment]
Thanks @dimbleby - yes - I have already created a bug inside torch repo: https://github.com/pytorch/pytorch/issues/91313
